dm/worker/config.go

```diff
@@ -239,6 +247,46 @@ func (c *Config) configFromFile(path string) error {
 	return c.verify()
 }
 
+// adjustFlavor adjusts flavor through querying from given database
+func (c *Config) adjustFlavor() error {
+	if c.Flavor != "" {
+		if c.Flavor != mysql.MariaDBFlavor && c.Flavor != mysql.MySQLFlavor {
+			return terror.ErrNotSupportedFlavor.Generate(c.Flavor)
+		}
+		return nil
+	}
+	// decrypt password
+	clone, err := c.DecryptPassword()
+	if err != nil {
+		return err
+	}
+	dbDSN := fmt.Sprintf("%s:%s@tcp(%s:%d)/?charset=utf8mb4&maxAllowedPacket=%d",
+		clone.From.User, clone.From.Password, clone.From.Host, clone.From.Port, *clone.From.MaxAllowedPacket)
+	conn, err := newBaseConn(dbDSN, nil, baseconn.DefaultRawDBConfig())
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	query := "SELECT @@version_comment"
```

Comment:
how about  move these code to a function like `GetFlavor` in pkg,  I need this function to do something

Reply:
OK